### PR TITLE
Add more actions to cell

### PR DESCRIPTION
**Pull request by YurongHe**
Added ToggleMoreWidgetAction to the right of the cell
Click it to show add code/markdown to before/after

image

### CODE REVIEW

File: src/sql/parts/notebook/cellViews/code.component.ts [hunk 1]
```diff
@@ -36,11 +42,22 @@ export const CODE_SELECTOR: string = 'code-component';
 })
 export class CodeComponent extends AngularDisposable implements OnInit {
 	@ViewChild('toolbar', { read: ElementRef }) private toolbarElement: ElementRef;
+	@ViewChild('moreactions', { read: ElementRef }) private moreactionsElement: ElementRef;
 	@ViewChild('editor', { read: ElementRef }) private codeElement: ElementRef;
 	@Input() cellModel: ICellModel;
+	@Input() set model(value: NotebookModel) {
```
Comment (kevcunnane) [inline src/sql/parts/notebook/cellViews/code.component.ts]: Minor: I know we put `@Input`s at the top normally, but for get/set we usually place below the constructor.

File: src/sql/parts/notebook/cellViews/codeActions.ts [hunk 2]
```diff
@@ -26,4 +29,50 @@ export class RunCellAction extends Action {
...
+					return;
+				}
+				let index = model.cells.findIndex((cell) => cell.id === model.activeCell.id);
+				if (index !== undefined && this.isAfter) {
+					index += 1;
+				}
+				model.addCell(this.cellType, index);
+			} catch (error) {
+				let message = getErrorMessage(error);
+				//ApiWrapper.showErrorMessage(message);
```
Comment (kevcunnane) [inline src/sql/parts/notebook/cellViews/codeActions.ts]: You will need to inject the `@INotificationService` like we do elsewhere. Then please uncomment and show the error.

File: src/sql/parts/notebook/cellViews/codeActions.ts [hunk 3]
```diff
@@ -26,4 +29,50 @@ export class RunCellAction extends Action {
...
+	public run(model: NotebookModel): TPromise<boolean> {
+		return new TPromise<boolean>((resolve, reject) => {
+			try {
+				if (!model) {
+					return;
+				}
+				model.deleteCell(model.activeCell);
+			} catch (error) {
+				let message = getErrorMessage(error);
+				//ApiWrapper.showErrorMessage(message);
```
Comment (kevcunnane) [inline src/sql/parts/notebook/cellViews/codeActions.ts]: Same comment as above - please update to show error.

**Comment by YurongHe** [comment]
Not needed.

**Pull request closed without merging**